src/libs/sys-util-rs/src/cpu.rs

```diff
@@ -4,6 +4,55 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+use std::collections::HashSet;
+
+use crate::sl;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Invalid CPU list {0}")]
+    InvalidCpuList(String),
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// Parse cpuset in form of "1,3,5" and "1-4,6,8".
+pub fn split_cpus(cpus_string: &str) -> Result<Vec<u32>> {
+    let mut cpu_list = HashSet::new();
+
+    if !cpus_string.is_empty() {
+        let cpus_string_split: Vec<&str> = cpus_string.split(',').collect();
```

Comment:
There's no need to `collect()` here, you can do the for loop directly over the `split` iterator.

Reply:
fixed